src/Images/ComponentSelector.jsx

```diff
@@ -19,6 +19,27 @@ function ComponentSelector({ service }) {
     (state) =>
       state.context.images.actorContext.get(name).image?.imageType.components
   )
+  // re-render when click on check-boxes
+  const componentVisibilities0 = useSelector(
+    service,
+    (state) =>
+      state.context.images.actorContext.get(name).componentVisibilities[0]
+  )
+  const componentVisibilities1 = useSelector(
+    service,
+    (state) =>
+      state.context.images.actorContext.get(name).componentVisibilities[1]
+  )
+  const componentVisibilities2 = useSelector(
+    service,
+    (state) =>
+      state.context.images.actorContext.get(name).componentVisibilities[2]
+  )
```

Comment:
This is an awesome suggestion. Thanks Paul. 
With the updates made in 7bb8fb407cad874f31e569f2159226575eb6436d, I think this has automatically been resolved. This is what I would do (future reminder if I need it):
```
  const compareComponentVisibilities = (prevComponent, nextComponent) =>
    prevComponent == nextComponent

  //re-render when click on check-boxes
  const componentVisibilities = useSelector(
    service,
    (state) =>
      state.context.images.actorContext.get(name).componentVisibilities,
    compareComponentVisibilities
  )

  // re-render when click on tabs
  const selectedComponent = useSelector(
    service,
    (state) => state.context.images.actorContext.get(name).selectedComponent
  )
```

Reply:
Thanks for catching that Paul! This has been fixed in [dc19d32](https://github.com/InsightSoftwareConsortium/itk-viewer-bootstrap-ui/pull/47/commits/dc19d322e3ef5d778a2ff35852c250555293a711). I tried to create a comparison function there to use as a third argument for `useSelector`, but I wasn't successful in it. It would be nice to use `shallowEqual`, but at this point I think we need to bump xstate up to be able to do so.